### ISSUE TITLE
Port polysemy-plugin and research/improve its usability

### ISSUE THREAD

**Issue by arybczak** [haskell-effectful/effectful#54]
It'd be great to port the `polysemy-plugin` to `effectful` to have the option to reduce ambiguity issues if a lot of parametrized effects are in use.

I briefly looked at the source and making it work with effectful should be fairly trivial, but there is a question of its usability. From looking at [polysemy issue tracker](https://github.com/polysemy-research/polysemy/issues?q=plugin+sort%3Acreated-desc) there were multiple tickets about the plugin behaving weirdly, though it looks like recently most (all?) of them were fixed.

Anyone would like to tackle this? I'll not deal with it myself before the initial release.

**Comment by sproott** [comment]
So I've ported the plugin and its tests. Needed to comment out a few tests though because of features that are not in `effectful`. 

The tests all run successfully, so I assume the plugin works as expected. To be honest, I have no idea how the plugin works, I just know how `effectful` works, so I just followed the compiler errors until everything was fine.

[The PR](https://github.com/arybczak/effectful/pull/55)
